ClemBot.Bot/bot/cogs/custom_tag_prefix_cog.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import typing as t
+
+import discord
+import discord.ext.commands as commands
+
+import bot.extensions as ext
+from bot.consts import Claims, Colors
+
+log = logging.getLogger(__name__)
+
+DEFAULT_TAG_PREFIX = '$'
+
+class CustomTagPrefixCog(commands.Cog):
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @ext.group(case_insensitive=True, invoke_without_command=True, aliases=['tagprefixs'])
+    @ext.long_help(
+        'Lists the current tag prefix or configures the command prefix that the bot will respond too'
+    )
+    @ext.short_help('Configure a custom command tag prefix')
+    @ext.example(('tagprefix', 'tagprefix ?', 'tagprefix >>'))
+    async def tagprefix(self, ctx, *, tagprefix: t.Optional[str] = None):
+        # get_prefix returns two mentions as the first possible prefixes in the tuple,
+        # those are global so we dont care about them
+        tag_prefixes = (await self.bot.get_tag_prefix(ctx.message))
+
+        if not tagprefix:
+            embed = discord.Embed(title='Current Active Tag Prefixes',
+                                  description=f'```{", ".join(tag_prefixes)}```',
+                                  color=Colors.ClemsonOrange)
+
+            return await ctx.send(embed=embed)
+
+        if tagprefix in await self.bot.get_tag_prefix(ctx.message):
```

Comment:
Shouldn't it be ... `get_tag_prefix(ctx)` instead? (although this will work because a `discord.Message` does have a `.guild` attribute)

Reply:
I followed the same code structure as `custom_prefix_cog.py` which used `get_prefix(ctx.message)`. But if you ask I can change it to just `ctx` and check if it works.

Reply:
I mean it is up to you and J

Reply:
either one works, Message was chosen at the time as i felt it semantically made more sense to see that the prefix class took a Message a dependency... becuase well, it does